tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -2385,6 +2412,7 @@ Status StorageManager::get_sorted_uris(
 
 Status StorageManager::get_uris_to_vacuum(
```

Comment:
In response to Stavros's comment: I think this is the problematic routine.

Consider the scenario where we have a vacuum file with timestamps `1`-`10` which contains a fragment written at a timestamp for each discrete number in the range. This vacuum file would intersect with a user-set vacuum between `5`-`7`.

In this scenario, we would place the vacuum file on `vac_uris` (L2429). On L2442, we read the contents and would put all `10` fragments on `to_vacuum_vec` (L2447). This is a mistake, we only want to vacuum fragments `5`-`7`. We definitely need timestamp comparison logic on `it->second` around L2447.

This leave us with an interesting problem: if we vacuum fragments `5`-`7`, we can't delete the vacuum file because it contains non-vacuumed fragments `1`-`4` and `8`-`10`. Stavros and I discussed two approaches:
1. Delete the vacuum and re-write a new vacuum file with only the non-vacuumed fragments.
2. Leave the vacuum file, and only delete it when all fragments have been vacuumed or will be vacuumed in the current operation.

We believe that #2 is the simpler and more robust approach. This routine will need to be modified so that the caller is aware of all fragments in each vacuum file to determine if they should delete the vacuum file. They'll need to use the VFS to check for the existence of the non-vacuumed fragments.

Reply:
In response to @stavrospapadopoulos comment, I believe this is a non issue. Vac files also have a begin and end timestamp and we filter those with that timestamp. All fragments in that vacuum file have start/end times between the boundaries of the file's start/end time. I think the "C API: Test vacuuming and timestamps" test should be sufficient to validate this.

Reply:
To reply to @joe-maley 's comment. In your example, the vacuum file will have start timestamp 1 and end timestamp 10, it will not be selected for vacuuming if the user specifies timestamps 5-7. So in the current implementation, we won't partially vacuum .vac files. I believe this makes sense since I doubt someone will consolidate from timestamps 2-7 and then later decide to only vacuum from timestamps 3-4. Do we really see a scenario where the user would need to partially vacuum is the question?

Reply:
You're right, the current implementation does not perform partial vacuuming. I'll let Stavros weigh in on the requirements. The current interface is is a leaky abstraction if the user must have knowledge about how we organize our vacuum files.

Reply:
@KiterLuc you are right, thanks for pointing this out! So that should be fine for now. 

Having said that, I do see a scenario where someone may wish to consolidate all fragments (e.g., hundreds) for performance, but retain the old ones for time traveling. I can see a case where the user may want to vacuum only a subset of those hundreds of consolidated fragments. We may want to tweak the behavior in the future to support "partial" vacuuming, but let's punt for now as we have some other improvements in mind regarding journaling the consolidated fragments to address some edge concurrency issues.